datasetinsights/datasets/groceries_real.py

```diff
@@ -227,22 +228,31 @@ def _preprocess_dataset(data_path, version):
             Return:
                 Path of the dataset files.
             """
-        if GroceriesReal.is_dataset_files_present(data_path):
-            return data_path
 
         archive_file = Path(data_path) / GroceriesReal.ARCHIVE_FILES[version]
-        if not archive_file.exists():
+        if archive_file.exists():
+            file_descriptor = os.open(archive_file, os.O_RDONLY)
+
+            try:
+                fcntl.flock(file_descriptor, fcntl.LOCK_EX)
+                unarchived_path = Path(data_path) / GroceriesReal.SUBFOLDER
+                if not GroceriesReal.is_dataset_files_present(unarchived_path):
```

Comment:
In general, nested if are harder to read and understand. Why do we prefer nested if over the current master which flattened these checks?